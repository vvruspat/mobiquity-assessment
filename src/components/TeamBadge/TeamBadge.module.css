.team {
    text-transform: uppercase;
    color: var(--secondary-color);
    letter-spacing: 2px;
}