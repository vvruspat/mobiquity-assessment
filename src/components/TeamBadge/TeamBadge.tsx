import { Badge } from "../Badge";
import { Text } from "../Text";
import s from "./TeamBadge.module.css";

type TeamBadgeProps = {
  team: string;
};

export const TeamBadge = ({ team }: TeamBadgeProps) => {
  return (
    <Badge badgeStyle="primary" size="m">
      <div className={s.team}>
        <Text size="s" weight="regular">
          {team}
        </Text>
      </div>
    </Badge>
  );
};
