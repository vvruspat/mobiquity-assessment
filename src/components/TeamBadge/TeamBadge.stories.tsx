import { ComponentStory, ComponentMeta } from "@storybook/react";

import { TeamBadge } from "./TeamBadge";
import "../../index.css";

export default {
  title: "Components/TeamBadge",
  component: TeamBadge,
} as ComponentMeta<typeof TeamBadge>;

const Template: ComponentStory<typeof TeamBadge> = (args) => (
  <TeamBadge {...args} />
);

export const TeamBadgeTemplate = Template.bind({});

TeamBadgeTemplate.args = {
  team: "Red bull",
};
