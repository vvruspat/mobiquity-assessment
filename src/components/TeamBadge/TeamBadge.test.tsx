import { render, screen } from "@testing-library/react";
import { TeamBadge } from "./TeamBadge";

test("renders team badge", () => {
  render(<TeamBadge team={"Red bull"} />);

  const el = screen.getByTestId("badge");
  expect(el).toHaveTextContent("Red bull");
  expect(el).toMatchSnapshot();
});
