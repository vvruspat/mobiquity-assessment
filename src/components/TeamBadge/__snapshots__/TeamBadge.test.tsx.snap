// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team badge 1`] = `
<div
  class="badge badge_m badge_style_primary badge_align_center"
  data-testid="badge"
>
  <div
    class="team"
  >
    <span
      class="text size_s regular"
      data-testid="text"
    >
      Red bull
    </span>
  </div>
</div>
`;
